### PR TITLE
Fix issue 3166

### PR DESCRIPTION
**Pull request by andyfengHKU**
I need to check this against CI first.

### CODE REVIEW

File: src/include/storage/store/rel_table.h [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 #include "storage/store/table.h"
 
 namespace kuzu {
 namespace storage {
 
 struct RelTableReadState : public TableReadState {
+    // TODO(Xiyang): rename me
     common::RelDataDirection direction;
 
+    // This direction is not
```
Comment (ray6080) [inline src/include/storage/store/rel_table.h:15]: Complete the sentence here.

File: src/function/struct/struct_pack_function.cpp [hunk 2]
```diff
@@ -65,6 +76,36 @@ void StructPackFunctions::execFunc(const std::vector<std::shared_ptr<ValueVector
     }
 }
 
+void StructPackFunctions::undirectedRelPackExecFunc(
+    const std::vector<std::shared_ptr<ValueVector>>& parameters, ValueVector& result, void*) {
+    KU_ASSERT(parameters.size() > 1);
+    // Force copy of
```
Comment (ray6080) [inline src/function/struct/struct_pack_function.cpp:82]: Complete the sentence.

File: src/function/struct/struct_pack_function.cpp [hunk 3]
```diff
@@ -33,6 +33,17 @@ void StructPackFunctions::compileFunc(FunctionBindData* /*bindData*/,
     }
 }
```
Comment (ray6080) [inline src/function/struct/struct_pack_function.cpp:35]: Better clearly state the implicit assumption of `src` being 0 and `dst` being 1 here.

File: src/include/expression_evaluator/node_rel_evaluator.h [hunk 4]
```diff
@@ -4,42 +4,54 @@
...
 
 namespace kuzu {
-namespace main {
-class ClientContext;
-}
-
 namespace evaluator {
 
-class NodeRelExpressionEvaluator final : public ExpressionEvaluator {
+class PatternExpressionEvaluator : public ExpressionEvaluator {
```
Comment (ray6080) [inline src/include/expression_evaluator/node_rel_evaluator.h:9]: Can u keep `final`?

File: src/include/expression_evaluator/node_rel_evaluator.h
Comment (ray6080) [inline src/include/expression_evaluator/node_rel_evaluator.h:1]: Let's also rename the file to `pattern_evaluator.h`.

**Review by ray6080 — APPROVED** [review]
I don't think it's a good design to fill the flip vector inside rel table scan. the reason is simply because that info is passed down from operator, while not relying on anything inside storage. Better move the filling of flip vector into operators.

File: src/function/struct/struct_pack_function.cpp (hunk 3 above)
Comment (andyfengHKU) [reply src/function/struct/struct_pack_function.cpp:35]: I'll try to do this in a more principled way.

**Pull request merged**